### PR TITLE
ci: split workflow into `build.yml` and `linting.yml`

### LINKED ISSUE #42: Split the current workflow into `build.yml` and `linting.yml` files
See the workflows [here](https://github.com/ANNUBS/annubes/tree/main/.github/workflows) as an example, which were adapted from one of the latest versions of the [eScience Center Python package template](https://github.com/NLeSC/python-template).

### PR DESCRIPTION
**Pull request by gcroci2**
Note that `setup.py` file can be improved quite a lot, but we will do it in #51

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+jobs:
+  build:
+    if: github.event.pull_request.draft == false
+    name: Build for (${{ matrix.python-version }}, ${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+        fail-fast: false
+        matrix:
+            os: ["ubuntu-latest"]
+            python-version: ["3.10", "3.11", "3.12"]
```
Comment (DaniBodor) [inline .github/workflows/build.yml:38]: shall we stick to a single python version to decrease the (environmental) overhead on this?

File: .github/workflows/linting.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      - name: Python info
+        shell: bash -e {0}
+        run: |
+          which python
+          python --version
+      - name: Check linting and formatting using ruff
+        run: |
+          python3 -m pip install ruff
+          ruff check || (echo "Please ensure you have the latest version of ruff (`ruff -V`) installed locally." && (exit 1))
+          ruff format --check || (echo "Please ensure you have the latest version of ruff (`ruff -V`) installed locally." && (exit 1))
```
Comment (DaniBodor) [inline .github/workflows/linting.yml]: empty final line was missing

```suggestion
          ruff format --check || (echo "Please ensure you have the latest version of ruff (`ruff -V`) installed locally." && (exit 1))

```

**Review by DaniBodor — APPROVED** [review]
Good idea to split these.
Agreed regarding setup.py (that it can be improved and that it can be in a separate PR)

File: .github/workflows/build.yml (hunk 1 above)
Comment (gcroci2) [reply .github/workflows/build.yml:38]: I would be in favor of keeping 3.11 and 3.12 only, but the original repo uses 3.10, so we need to keep it for now (we can propose to the authors later to remove the older 3.10). It usually good to maintain at least a couple of python versions (the latest ones), to not force the users too much in using only 1 specific python version. I'm also aware about python 3.13, but it is still under development and the final is scheduled for October (see [here](https://peps.python.org/pep-0719/)). Better to stick to more stable and widely supported versions.

**Pull request merged**